src/test/java/io/cdap/plugin/servicenow/source/ServiceNowInputFormatTest.java

```diff
@@ -0,0 +1,311 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package io.cdap.plugin.servicenow.source;
+
+import io.cdap.plugin.servicenow.restapi.RestAPIResponse;
+import io.cdap.plugin.servicenow.source.apiclient.ServiceNowTableAPIClientImpl;
+import io.cdap.plugin.servicenow.source.util.SourceQueryMode;
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.oltu.oauth2.client.OAuthClient;
+import org.apache.oltu.oauth2.client.URLConnectionClient;
+import org.apache.oltu.oauth2.client.response.OAuthJSONAccessTokenResponse;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.internal.AssumptionViolatedException;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ServiceNowTableAPIClientImpl.class, ServiceNowBaseSourceConfig.class, ServiceNowMultiSource.class,
+        HttpClientBuilder.class, RestAPIResponse.class})
+public class ServiceNowInputFormatTest {
+
+  private static final String CLIENT_ID = "clientId";
+  private static final String CLIENT_SECRET = "clientSecret";
+  private static final String REST_API_ENDPOINT = "https://ven05127.service-now.com";
+  private static final String USER = "user";
+  private static final String PASSWORD = "password";
+  private static final Logger LOG = LoggerFactory.getLogger(ServiceNowInputFormat.class);
+  private ServiceNowSourceConfig config;
+
+  @Before
+  public void initializeTests() {
+    try {
+      config = Mockito.spy(new ServiceNowSourceConfig("Reference Name", "Query Mode",
+                                          "Product Catalog", "tablename", "pc_hardware_cat_item",
+                                          CLIENT_ID, CLIENT_SECRET,
+                                          REST_API_ENDPOINT, USER, PASSWORD, "Actual", "2012-12-31", "2021-12-31"));
+    } catch (AssumptionViolatedException e) {
+      LOG.warn("Service Now batch Source tests are skipped. ");
```

Comment:
why would AssumptionViolatedException be thrown?

Reply:
This exception was previously being thrown when the user-provided credentials were taken from the System properties specified in the VM arguments.
This try-catch block is no longer required as dummy credentials are being provided now. cf1e022323014e4a0be081ac6a3189c04770526a